nexus/src/app/rack.rs

```diff
@@ -985,13 +1030,11 @@ impl super::Nexus {
             },
         };
         let sa = self.get_any_sled_agent(opctx).await?;
-        sa.add_sled_to_initialized_rack(&req).await.map_err(|e| {
-            Error::InternalError {
-                internal_message: format!(
-                    "failed to add sled with baseboard {:?} to rack {}: {e}",
-                    sled.baseboard, allocation.rack_id
-                ),
-            }
+        sa.sled_add(&req).await.map_err(|e| Error::InternalError {
+            internal_message: format!(
+                "failed to add sled with baseboard {:?} to rack {}: {e}",
+                sled, allocation.rack_id
+            ),
```

Comment:
No worse than before, but this would incorrectly report a transient (e.g., network) issue as a 500 when it should be a 503.

Reply:
Would you prefer I change this to a 503 or be more specific and check to see which error is returned? I'm not sure the sled-agent differentiates between a 500 and 503 for this endpoint and in most cases 500 is probably better.

Reply:
Ideally we'd check which error was returned.  It might not even be an error from the sled agent.  For example right now if we got DNS resolution error or a TCP connection failure of some kind, we'd report that as a 500, but that should probably be a 503 (more on this [here](https://github.com/oxidecomputer/omicron/blob/main/docs/http-status-codes.adoc#34-server-errors)).

I'm not sure it's worth dealing with in this PR, since the problem predates this and I think the fix is probably more systemic.  We probably want to make a helper function that we can use in a bunch of places (similar in spirit to [this one](https://github.com/oxidecomputer/omicron/blob/de152304c3f551914db05bda1ddbc4b139e1f766/clients/dns-service-client/src/lib.rs#L27-L83) and then use it everywhere we make a Progenitor call (similar to the way we use `public_error_from_diesel` every time we make a database call...though I just noticed that one seems to have the same problem!)